### PR TITLE
exchange.handleNetworkParameter

### PR DESCRIPTION
**Pull request by samgermain**
I know that there's already `handleNetworkCodeAndParams`, but it doesn't handle default networks, so I think this is a better solution. I would have updated `handleNetworkCodeAndParams` instead of writing a new method, but this method requires a `code` argument, so that would interfere with the exchanges that are calling `handleNetworkCodeAndParams` already

## TS

2024-02-09T03:04:25.653Z
Node.js: v18.12.0
CCXT v4.2.39

```
% mexc handleNetworkParameter USDT                        
mexc.handleNetworkParameter (USDT)
2024-02-09T03:04:29.117Z iteration 0 passed in 1 ms

[ 'TRC20', {} ]
2 objects
2024-02-09T03:04:29.117Z iteration 1 passed in 1 ms
```
```
% mexc handleNetworkParameter USDT '{"network": "ERC20"}'
mexc.handleNetworkParameter (USDT, [object Object])
2024-02-09T03:04:55.390Z iteration 0 passed in 0 ms

[ 'ERC20', {} ]
2 objects
2024-02-09T03:04:55.390Z iteration 1 passed in 0 ms
```
```
% mexc handleNetworkParameter LTC                        
mexc.handleNetworkParameter (LTC)
2024-02-09T03:05:09.938Z iteration 0 passed in 0 ms

[ undefined, {} ]
2 objects
2024-02-09T03:05:09.938Z iteration 1 passed in 0 ms
```

## Python

Python v3.9.6
CCXT v4.2.39

```
% py mexc handleNetworkParameter USDT   
mexc.handleNetworkParameter(USDT)
['TRC20', {}]
```
```
% py mexc handleNetworkParameter USDT '{"network": "ERC20"}'
mexc.handleNetworkParameter(USDT,{'network': 'ERC20'})
['ERC20', {}]
```
```
% py mexc handleNetworkParameter LTC                        
mexc.handleNetworkParameter(LTC)
[None, {}]
```

### CODE REVIEW

File: ts/src/base/Exchange.ts [hunk 1]
```diff
@@ -5982,6 +5982,14 @@ export default class Exchange {
     parseGreeks (greeks, market: Market = undefined): Greeks {
         throw new NotSupported (this.id + ' parseGreeks () is not supported yet');
     }
+
+    handleNetworkParameter (code: string, params = {}) {
```
Comment (carlosmiei) [inline ts/src/base/Exchange.ts:5986]: As is `handleNetworkParameter` is not very useful, it should be using `networkCodeToId` instead

File: ts/src/base/Exchange.ts [hunk 2]
```diff
@@ -5982,6 +5982,14 @@ export default class Exchange {
...
+        const defaultNetworks = this.safeDict (this.options, 'defaultNetworks');
+        const defaultNetwork = this.safeString (defaultNetworks, code);
+        const network = this.safeStringN (params, [ 'network', 'defaultNetwork', 'networkCode' ], defaultNetwork);
```
Comment (carlosmiei) [inline ts/src/base/Exchange.ts:5989]: then. you read like 3 different params, but only omit 1, why?

**Comment by carlosmiei** [comment]
we have already handleNetworkCodeAndParams so this method is unnecessary

**Comment by carlosmiei** [comment]
As I said previously, this method is confusing (because usually, handelX methods are meant to build the request and is similar to handleNetworkCode), so I don't think we should add it. We might implement a mexc-specific solution if needed

**Comment by samgermain** [comment]
> As I said previously, this method is confusing (because usually, handelX methods are meant to build the request and is similar to handleNetworkCode), so I don't think we should add it. We might implement a mexc-specific solution if needed

Yes it is similar to `handleNetworkCode` I explained above why I didn't just edit `handleNetworkCode

File: ts/src/base/Exchange.ts (hunk 1 above)
Comment (samgermain) [reply ts/src/base/Exchange.ts:5986]: > As is `handleNetworkParameter` is not very useful, it should be using `networkCodeToId` if network is available

`networkCodeToId` has a completely different purpose, it gets the exchange specific networkId based on the CCXT network code. `handleNetworkParameter` provides a default network code from `options` if a `networkCode` is not provided in params

**Comment by ttodua** [comment]
guys, i have been doing the detailed handling of network codes and ids in my past PR, with too much time in research / details /updates, so i think, Sam , you might not need to spend time on this, give me few days and I'll try to revive my PR and simplify it

**Comment by ttodua** [comment]
i will also review this PR to see if you have been doing something else

**Pull request closed without merging**